tests/ut/ir/transforms/test_lower_host_tensor_collectives.py

```
@@ -0,0 +1,151 @@
+        def host_orch(self):
+            data_buf = pld.alloc_window_buffer(1024)
+            signal_buf = pld.alloc_window_buffer(4)
+            data = pld.window(data_buf, [256], dtype=pl.FP32)
+            signal = pld.window(signal_buf, [1], dtype=pl.INT32)
+            self.chip_orch(data, device=0)
+            self.chip_orch(data, device=1)
+            pld.tensor.allreduce_(data, signal, op=pld.ReduceOp.Sum)
+            return 0
+
+    program = passes.materialize_comm_domain_scopes()(P)
+    with pytest.raises(Exception, match="signal shape\\[0\\].*participating device count"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)